SNOW-11: PortalNavigationProviderBase (#106)
Lombiq/Helpful-Libraries: Lombiq.HelpfulLibraries/Docs/Navigation.md

# Navigation Documentation

## NavigationProviderBase

> An abstract base class for creating reducing boilerplate in `INavigationProvider` for checking the name parameter and if the user is authenticated. 

Reviewer: ```suggestion
An abstract base class to reduce the boilerplate in `INavigationProvider` implementations for matching the name and authenticating the user.
```

# Navigation Documentation

> ## PortalNavigationProviderBase

Reviewer 2: I commented it to the base theme, but I'll leave it here too:

> I think the word Portal is too specific. With this widget, you are providing the ability to render navigation anywhere. Maybe SiteNavigationWidgetFilter would be more appropriate.

Author: I'm providing for the navigation providers with the name "portal". I can make an even more abstract base class if we anticipate the need for other navigation types, but I thought we can do that when needed. Am I missing some consideration?

Author: nvm, I've created a base class with the overridable NavigationName

Reviewer 2: My issue is with the class name and basically the widget name. It can be any navigation and the name "portal" in the class name indicates something more specific. The navigation name itself is another story, but that can be generic, too. The word "portal" is coming from SNOW in the first place.

Author: Okay, now I'm confused. wtf is a "portal" anyway? I'm only used to _thinking with portals_ :)
Wikipedia says "a specially designed website that brings information from diverse sources, like emails, online forums and search engines, together in a uniform way". Is that applicable to SNOW?

Author: I renamed the navigation type "main" as I said in the other PR. It is most sensible this way. Similarly the provider becomes `MainMenuNavigationProviderBase` but the widget is simply MenuWidget as it doesn't know where its MenuItems come from (though I still dunno where else we could use that widget).

> An abstract base class for creating home page menu structure using the `portal` navigation name.If you use the _Lombiq.BaseTheme_ it automatically displays the generated menu as a widget in the _Navigation_ zone.

Reviewer 3: ```suggestion
An abstract base class for creating home page menu structure using the `portal` navigation name. If you use the _Lombiq.BaseTheme_ it automatically displays the generated menu as a widget in the _Navigation_ zone.
```
